proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,238 @@
+# Opt-in experience for PrivateAssets and ExcludeAssets options are independent
+
+- Author Name <https://github.com/erdembayar>
+- Start Date 2022-12-07
+- GitHub Issue <https://github.com/NuGet/Home/issues/6938>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4976>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Currently asset consumption via `PrivateAssets` option experience for assets from transitive package in a project/package is not deterministic for parent consuming project, here `PrivateAssets` option calculation is not independent from `IncludeAssets/ExcludeAssets` option, if an asset is not consumed in current project then it doesn't flow up the regardless of `PrivateAssets` option.
+This proposal introduces new a boolean `ExcludedAssetsFlow` metadata to improve experience for `PrivateAssets` option to make it independent from `IncludeAssets/ExcludeAssets` option, it'll make assets specified in `PrivateAssets` will flow regardless of `IncludeAssets/ExcludeAssets` option.
```

Comment:
> it'll make assets specified in `PrivateAssets` will flow

Actually the opposite: All assets _not_ specified in `PrivateAssets` will flow when the new metadata is set.
I don't mean to be nit-picky, but given the inverting nature of the metadata and the confusion in this area already, we should probably be as precise and accurate as the problem space requires.

Reply:
You're right, I did the example below correctly, somehow messed up the explanation.